server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/LocalShardsBalancer.java

```diff
@@ -527,6 +531,128 @@ private void checkAndAddInEligibleTargetNode(RoutingNode targetNode) {
         }
     }
 
+    /**
+     * Triggers shard relocation for the shards that are submitted
+     * for tiering to go from local node pool to remote capable node
+     * pool.
+     */
+    public void tierShards() {
+        List<ShardRouting> shardsPendingTiering = new ArrayList<>();
```

Comment:
It might be better to break this down into replicas and primary.
We can run the operation on replicas first, and the migration for primary would handle the scenario of flipping replica to primary. 
(There would be a case where you have no replica - which is fine. But when you do, it is safer)